public/app/plugins/datasource/cloudwatch/query-runner/CloudWatchAnnotationQueryRunner.ts

```diff
@@ -10,18 +9,21 @@ import { CloudWatchRequest } from './CloudWatchRequest';
 
 // This class handles execution of CloudWatch annotation queries
 export class CloudWatchAnnotationQueryRunner extends CloudWatchRequest {
-  constructor(instanceSettings: DataSourceInstanceSettings<CloudWatchJsonData>, templateSrv: TemplateSrv) {
-    super(instanceSettings, templateSrv);
+  constructor(
+    instanceSettings: DataSourceInstanceSettings<CloudWatchJsonData>,
+    templateSrv: TemplateSrv,
+    queryFn: (request: DataQueryRequest<CloudWatchQuery>) => Observable<DataQueryResponse>
+  ) {
+    super(instanceSettings, templateSrv, queryFn);
   }
 
   handleAnnotationQuery(
     queries: CloudWatchAnnotationQuery[],
     options: DataQueryRequest<CloudWatchQuery>
   ): Observable<DataQueryResponse> {
-    return this.awsRequest(this.dsQueryEndpoint, {
-      from: options.range.from.valueOf().toString(),
-      to: options.range.to.valueOf().toString(),
```

Comment:
interesting that we don't need to pass from and to anymore? Are those already in options?

Reply:
[`super.query` handles this](https://github.com/grafana/grafana/blob/d410ef343946c8afcd0a1792bdba6fe6f6388e6b/packages/grafana-runtime/src/utils/DataSourceWithBackend.ts#L195-L196) because the [`range` is part of the request options we pass in](https://github.com/grafana/grafana/blob/d410ef343946c8afcd0a1792bdba6fe6f6388e6b/packages/grafana-runtime/src/utils/DataSourceWithBackend.ts#L132)